pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/TransformerWorker.scala

```diff
@@ -0,0 +1,95 @@
+package uk.ac.wellcome.platform.transformer.calm
+
+import akka.{Done, NotUsed}
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import com.amazonaws.services.sqs.model.Message
+import uk.ac.wellcome.bigmessaging.BigMessageSender
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.sqs.SQSStream
+import uk.ac.wellcome.models.work.internal.TransformedBaseWork
+import uk.ac.wellcome.storage.store.VersionedStore
+import uk.ac.wellcome.storage.{Identified, Version}
+
+import scala.concurrent.Future
+
+sealed abstract class CalmWorkerError(msg: String) extends Exception(msg)
+case class DecodeKeyError(msg: String) extends CalmWorkerError(msg)
+case class StoreReadError(msg: String) extends CalmWorkerError(msg)
+case class TransformerError(msg: String) extends CalmWorkerError(msg)
+case class MessageSendError(msg: String) extends CalmWorkerError(msg)
+
+/**
+  * A transformer worker:
+  * - Takes an SQS stream that emits VHS keys
+  * - Gets the record of type `In`
+  * - Runs it through a transformer and transforms the `In` to `TransformedBaseWork`
+  * - Emits the message via `BigMessageSender` to an SNS topic
+  */
+trait TransformerWorker[In, SenderDest] {
+  type StreamMessage = (Message, NotificationMessage)
+  type Result[T] = Either[Throwable, T]
+  type StoreKey = Version[String, Int]
+
+  val name: String
```

Comment:
How about this? It generally ends up being the class name in all our workers anyway (and could override with something different if required):

```suggestion
  def name: String = this.getClass.getSimpleName
```